WordPress/gutenberg
# Placeholder component: Replace upload icon with text when rendering labels

## What?

- Renders the text label instead of the upload icon for the `Placeholder` component when the `Show button text labels` preference is enabled. 
- Fixes a minor UI issue with the block mover when labels are enabled.
- Reduces specificity for the `Featured image block`'s styles so that it can be overwritten when labels are to be rendered for the placeholder. This change overlaps with a change in https://github.com/WordPress/gutenberg/pull/44286, but is required for this PR as well. Credit to the PR author.
## Why?

- Check out https://github.com/WordPress/gutenberg/issues/44115#issuecomment-1251269740

## How?


## Testing Instructions
Since this PR touches a couple of widely used components' styles, it should be tested thoroughly. 
1. Open the post editor.
2. Go to `Options > Preferences > General` and enable `Show button text labels`.  See [this comment](https://github.com/WordPress/gutenberg/issues/44115#issuecomment-1256460097) for visuals of this option. 
3. Insert a `Post Featured Image` block. 
4. Move your focus on the block by clicking on it if you don't see any button or block controls for it.
5. You _should_ see a button with the label `Add a featured image` instead of the circular upload button.

## Screenshots or screencast 
|Before|After|
|------|------|
|![Screenshot 2022-09-25 at 9 41 04 PM](https://user-images.githubusercontent.com/59014930/192153921-7db3935d-6db2-42ee-90d6-b65d2a24a4cf.png)|<img width="855" alt="Screenshot 2022-09-29 at 4 06 35 AM" src="https://user-images.githubusercontent.com/59014930/192900713-298748de-c3dc-4862-afb9-f97f74d84b13.png">|
|![Screenshot 2022-09-26 at 1 38 44 AM](https://user-images.githubusercontent.com/59014930/192163154-e7b7433b-1cfb-4789-a5a5-29b05d7c4059.png)|<img width="793" alt="Screenshot 2022-09-29 at 4 07 59 AM" src="https://user-images.githubusercontent.com/59014930/192900869-0bc441d2-163b-453b-b716-1f58bb97c1a3.png">|
